packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,566 @@
+   * the SimpleTransactionBuilder, the method will fail to get the transaction data
+   * @param {string} txId The transaction id.
+   * @returns {Tx} The transaction details of the given transaction id.
+   */
+  async getTransactionData(txId: string): Promise<Tx> {
+    try {
+      const transactionCommittedDetailsRequest: TransactionCommittedDetailsRequest = {
+        intent_hash: txId,
+        opt_ins: {
+          raw_hex: true,
+        },
+      }
+      const transactionCommittedDetailsResponse: TransactionCommittedDetailsResponse =
+        await this.gatewayApiClient.transaction.innerClient.transactionCommittedDetails({
+          transactionCommittedDetailsRequest: transactionCommittedDetailsRequest,
+        })
+      if (
+        transactionCommittedDetailsResponse.transaction.raw_hex !== undefined &&
+        transactionCommittedDetailsResponse.transaction.confirmed_at !== null &&
+        transactionCommittedDetailsResponse.transaction.confirmed_at !== undefined &&
+        transactionCommittedDetailsResponse.transaction.intent_hash !== undefined
+      ) {
+        const transaction: Tx = await this.convertTransactionFromHex(
+          transactionCommittedDetailsResponse.transaction.raw_hex,
+          transactionCommittedDetailsResponse.transaction.intent_hash,
+          transactionCommittedDetailsResponse.transaction.confirmed_at,
+        )
+        return transaction
+      } else {
+        throw new Error('Incomplete transaction data received')
+      }
+    } catch (error) {
+      throw new Error('Failed to fetch transaction data')
+    }
+  }
+
+  /**
+   * Helper function to convert a transaction in hex, returned by the gateway to a Tx type
+   * @param transaction_hex - The raw_hex returned by the gateway for a transaction id
+   * @param confirmed_at - The confirmed_at date for the transaction
+   * @param intent_hash - The transaction intent hash
+   * @returns a transaction in Tx type
+   */
+  async convertTransactionFromHex(transaction_hex: string, intent_hash: string, confirmed_at: Date): Promise<Tx> {
+    const binaryString = Buffer.from(transaction_hex, 'hex').toString('binary')
+    const transactionBinary = new Uint8Array(binaryString.split('').map((char) => char.charCodeAt(0)))
+    const transactionSummary = await LTSRadixEngineToolkit.Transaction.summarizeTransaction(transactionBinary)
+    const withdrawAccount = Object.keys(transactionSummary.withdraws)[0]
+    const depositAccount = Object.keys(transactionSummary.deposits)[0]
+    const withdrawResource = Object.keys(transactionSummary.withdraws[withdrawAccount])[0]
+    const withdrawAmount: number = transactionSummary.withdraws[withdrawAccount][withdrawResource].toNumber()
+
+    const transaction: Tx = {
+      from: [
+        {
+          from: withdrawAccount,
+          amount: baseAmount(withdrawAmount),
+          asset: { symbol: withdrawResource, ticker: withdrawResource, synth: false, chain: 'radix' },
```

Comment:
`chain` here is lowercase `radix` while the assets defined in `const.ts` use the `RadixChain` constant which has the chain set to uppercase `RADIX`.